addOns/fuzz/src/main/java/org/zaproxy/zap/extension/fuzz/FuzzAPI.java

```diff
@@ -0,0 +1,842 @@
+                try {
+                    locationStart =
+                            Integer.parseInt(fuzzLocation.split(FUZZ_LOCATION_SEPARATOR)[0]);
+                    locationEnd = Integer.parseInt(fuzzLocation.split(FUZZ_LOCATION_SEPARATOR)[1]);
+                } catch (IllegalArgumentException e) {
+                    return new ApiResponseElement("Illegal fuzzLocation input: " + fuzzLocation);
+                }
+                String payloadPath = getParam(params, PARAM_PAYLOAD_PATH, null);
+
+                String fuzzHeader = getParam(params, PARAM_FUZZ_REQUEST_LOCATION, null);
+                HttpMessageLocation.Location httpLocation;
+                if (fuzzHeader.toLowerCase().equals(MESSAGE_LOCATION_BODY)) {
+                    httpLocation = HttpMessageLocation.Location.REQUEST_BODY;
+                } else if (fuzzHeader.toLowerCase().equals(MESSAGE_LOCATION_HEADER)) {
+                    httpLocation = HttpMessageLocation.Location.REQUEST_HEADER;
+                } else {
+                    return new ApiResponseElement(
+                            "Invalid request Location: "
+                                    + fuzzHeader
+                                    + " only \"body\" or \"header\" allowed");
+                }
+                List<PayloadGeneratorMessageLocation<?>> fuzzLocations =
+                        createFuzzLocations(httpLocation, locationStart, locationEnd, payloadPath);
+                if (fuzzLocations == null) {
+                    LOGGER.debug("Failed to load fuzz locations");
+                    return new ApiResponseElement(
+                            "Error", "Failed to load fuzzLocations and payloads");
+                }
+                httpFuzzerMessageProcessors.add(RequestContentLengthUpdaterProcessor.getInstance());
+                HttpFuzzer httpFuzzerSimple =
+                        createFuzzer(
+                                recordHistory.getHttpMessage(),
+                                fuzzLocations,
+                                getOptions(),
+                                httpFuzzerMessageProcessors);
+                extension.runFuzzer(httpFuzzerHandler, httpFuzzerSimple);
+                assert httpFuzzerSimple != null;
+                return new ApiResponseElement(
+                        "fuzz.api.response.fuzzerid",
+                        Integer.toString(httpFuzzerSimple.getScanId()));
+            case ACTION_RESET_DEFAULT_HTTP_FUZZ_OPTIONS:
+                resetHttpFuzzerOptions();
+                return ApiResponseElement.OK;
+            case ACTION_SET_HTTP_FUZZ_OPTIONS:
+                setHttpFuzzerOptions(
+                        getParam(params, PARAM_MAX_ERRORS_ALLOWED, -1),
+                        getParam(params, PARAM_STRATEGY, null),
+                        getParam(params, PARAM_RETRIES, -1),
+                        getParam(params, PARAM_DELAY, -1),
+                        getParam(params, PARAM_THREADS, -1));
```

Comment:
Will threads -1 work?

Will all these work defaulted to -1?

Edit: I guess this is addressed at Line 403?